### PR TITLE
feat(preview-email): new definition

### PR DESCRIPTION
**Pull request by peterblazejewicz**
- definition type for `preview-email)
- tests

https://github.com/forwardemail/preview-email#usage

Thanks!

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. 
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/preview-email/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+    interface OpenOptions {
+        /** @default false */
+        wait?: boolean;
+    }
+
+    /**
+     * a function to build preview url from file path
+     */
+    interface UlrTransform {
```
Comment (sandersn) [inline types/preview-email/index.d.ts:55]:
```suggestion
    interface UrlTransform {
```

File: types/preview-email/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+         *  file path to a pug template file (defaults to preview-email's `template.pug` by default)
+         * - this is where you can pass a custom template for rendering email previews, e.g. your own stylesheet
+         */
+        template?: string;
+        /**
+         * a function to build preview url from file path
+         * - this is where you can customize the opened path to handle WSL to Windows transformation or build a http url if dir is served.
+         * @default (path) => 'file://[file path]'
+         */
+        urlTransform?: UlrTransform;
```
Comment (sandersn) [inline types/preview-email/index.d.ts:44]:
```suggestion
        urlTransform?: UrlTransform;
```

**Review by sandersn — CHANGES_REQUESTED** [review]
Noticed some typos, otherwise looks good.

**Review by sandersn — APPROVED** [review]

**Comment by peterblazejewicz** [comment]
Ready to merge 💋

**Pull request merged**